zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZkClientUriDomainMappingHelper.java

```diff
@@ -79,8 +79,17 @@ public ZkClientUriDomainMappingHelper(ZooKeeperServer zks) {
     }
 
     if (zks.getZKDatabase().getNode(rootPath) == null) {
-      throw new IllegalStateException(
-          "ZkClientUriDomainMappingHelper::ClientUriDomainMapping root path does not exist :" + rootPath);
+      try {
+        // Create a read-only root path, super user (admins) will need to manually set the correct
+        // ACL to root path in order to add clientUri-domain pairs to the mapping
+        zks.getZKDatabase().getDataTree().createNode(rootPath, new byte[0],
+            ZooDefs.Ids.READ_ACL_UNSAFE, -1L, -1,
+            0L, 0L);
+      } catch (Exception e) {
+        LOG.warn(
+            "ZkClientUriDomainMappingHelper::Failed to create client uri domain mapping root path node, exception: ",
```

Comment:
Don't we want to fail here or return from here atleast because i'm not sure what would happen if we try to add watches if we fail to create path itself.

Reply:
Good point. I was thinking about NodeExistException, but you are right, we should fail for other cases if there's exception occurring. Updated the code.